frontend/src/features/statistics/StatsContext.tsx

```diff
@@ -52,23 +52,26 @@ export const StatsLazyProvider = ({
    * Initialize the stats if they are empty or refresh them if they are
    * outdated.
    */
-  const getStats = useCallback(() => {
-    const currentTime = Date.now();
+  const getStats = useCallback(
+    (poll: string | undefined) => {
+      const currentTime = Date.now();
 
-    if (loading.current) {
-      return stats;
-    }
+      if (loading.current) {
+        return stats;
+      }
 
-    if (
-      stats.polls.length === 0 ||
-      currentTime - lastRefreshAt.current >= EXPIRATION_TIME
-    ) {
-      loading.current = true;
-      refreshStats();
-    }
+      if (
+        stats.polls.length === 0 ||
+        currentTime - lastRefreshAt.current >= EXPIRATION_TIME
```

Comment:
add a condition to force refresh when current `poll` is not present in `stats.polls` ?

Reply:
Addressed by https://github.com/tournesol-app/tournesol/pull/1830/commits/9df3994a527b3c68cfcd4fc738d79b3f5108ed54

I finally opted for a ref, instead of checking if the poll was present in `stats.polls`. This allows to handle the specific case where `getStats` is called two times in less than 4 seconds, a first time with a poll, a second time with no poll.